DNN Platform/Library/Entities/Portals/PortalController.cs

```diff
@@ -163,6 +163,12 @@ private static void CreateDefaultPortalRoles(int portalId, int administratorId,
             }
             string mappedHomeDirectory = String.Format(Globals.ApplicationMapPath + "\\" + homeDirectory + "\\").Replace("/", "\\");
 
+            if(Directory.Exists(mappedHomeDirectory))
+            {
+                message += string.Format(Localization.GetString("CreatePortalHomeFolderExists.Error"), homeDirectory);
+                throw new Exception(message);

```

Comment:
I'm not convinced about a basic `Exception`. At least `InvalidOpertionException`? Or maybe some type DNN-specific?